src/trainer.py

```diff
@@ -0,0 +1,125 @@
+# __author__ = "Vasudev Gupta"
+# __author_email__ = "7vasudevgupta@gmail.com"
+
+"""Simple Trainer for handling TensorFlow Model Training"""
+
+from dataclasses import asdict, dataclass
+
+import tensorflow as tf
+import wandb
+from tqdm.auto import tqdm
+
+@dataclass
+class Trainer:
+    lr: float
+    max_epochs: int = 2
+
+    logging_steps: int = 10
+    eval_steps: int = 10
+
+    use_tpu: bool = False
+
+    save_steps: int = 10
+    ckpt_dir: str = "checkpoints" # None
+
+    # wandb args
+    project_name: str = "gsoc-wav2vec2"
+    run_name: str = "finetuning"
```

Comment:
Ditto with using a `__init__()`.

Reply:
same doubt as above in `data_utils.py`.

Reply:
See my reply there.

Reply:
I will shift this to normal class as well & will make separate args for `Trainer`.